src/mode.ts

```diff
@@ -30,16 +48,23 @@ export function setupJavaScript(defaults:LanguageServiceDefaultsImpl): void {
 	);
 }
 
-function setupMode(defaults:LanguageServiceDefaultsImpl, modeId:string, language:Language): void {
+export function getEmitOutput(uri: monaco.Uri): monaco.Promise<any> {
+	return executeAfterSetup((uri: monaco.Uri) => {
+		return worker(uri)
+				.then((worker: TypeScriptWorker) =>
+					worker.getEmitOutput(uri.toString()))
+		}, null, [uri]);
+}
 
+function setupMode(defaults:LanguageServiceDefaultsImpl, modeId:string, language:Language): void {
 	let disposables: IDisposable[] = [];
 
-	const client = new WorkerManager(defaults);
-	disposables.push(client);
+	if (setupModes.indexOf(modeId) > -1) {
+		return;
+	}
 
-	const worker = (first: Uri, ...more: Uri[]): Promise<TypeScriptWorker> => {
-		return client.getLanguageServiceWorker(...[first].concat(more));
-	};
+	client = new WorkerManager(defaults);

```

Comment:
Two workers as we have it [now](https://github.com/Microsoft/monaco-typescript/blob/master/src/mode.ts#L37). Since they are scoped one is asked for JS files and the other for TS files